File: view_based_projects/06-3-BoardOfMessagesAppV2/app/src/test/java/com/example/boardofmessagesapp/repository/BoardRepositoryAddBoardTest.java
package com.example.boardofmessagesapp.repository;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.fail;

import org.junit.Before;
import org.junit.Test;

import java.util.HashMap;
import java.util.List;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

public class BoardRepositoryAddBoardTest {

    private CountDownLatch lock;
    private BoardRepository repo = new BoardRepository(true);
    private Board resBoard;

    @Before
    public void before() throws Exception {
        repo.reset(); // reset API
        lock = new CountDownLatch(1);
    }

    @Test
    public void testAddBoard() throws Exception {
        Board b = new Board();
        b.setName("my board");
        repo.addBoard(b, new BoardRepository.BoardCallback() {
            @Override
            public void onSuccess(Board board) {
                resBoard = board;
                lock.countDown();
            }

            @Override
            public void onError(String errorMessage) {
                fail();
                lock.countDown();
            }
        });

        lock.await(10, TimeUnit.SECONDS);
        assertEquals("my board", resBoard.getName());
    }
}

File: kotlin_projects/06-3-BoardOfMessageApp_kt2/app/src/test/java/com/example/boardofmessageapp_kt2/repository/common/BoardRepositoryAddBoardTest.kt
package com.example.boardofmessageapp_kt2.repository.common

import kotlinx.coroutines.runBlocking
import org.junit.Test

import org.junit.Assert.*
import org.junit.Before

class BoardRepositoryAddBoardTest {

    val boardRepository = BoardRepository(true)

    @Before
    fun before() {
        runBlocking { boardRepository.reset() }
    }

    @Test
    fun `test AddBoard`() =  runBlocking {
        val newBoard = Board(
            name = "new board",
            messages = listOf()
        )
        val board = boardRepository.addBoard(newBoard)
        assertEquals("new board", board.name)
        assertEquals(3, board.id)
        assertEquals(0, board.messages.size)
    }
}